src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -367,5 +373,86 @@ public static class AkkaClusterHostingExtensions
                 CreateAndRegisterSingletonProxy<TKey>(singletonName, singletonManagerPath, singletonProxySettings, system, registry);
             });
         }
+
+        /// <summary>
+        /// Configures a <see cref="ClusterClientReceptionist"/> for the <see cref="ActorSystem"/>
+        /// </summary>
+        /// <param name="builder">The builder instance being configured.</param>
+        /// <param name="name">Actor name of the ClusterReceptionist actor under the system path, by default it is /system/receptionist</param>
+        /// <param name="role">Checks that the receptionist only start on members tagged with this role. All members are used if empty.</param>
+        /// <returns>The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.</returns>
+        public static AkkaConfigurationBuilder WithClusterClientReceptionist(
+            this AkkaConfigurationBuilder builder,
+            string name = "receptionist",
+            string role = null)
+        {
+            builder.AddHocon(CreateReceptionistConfig(name, role), HoconAddMode.Prepend);
+            return builder;
+        }
+
+        internal static Config CreateReceptionistConfig(string name, string role)
+        {
+            const string root = "akka.cluster.client.receptionist.";
+            
+            var sb = new StringBuilder()
+                .Append(root).Append("name:").AppendLine(QuoteIfNeeded(name));
+            
+            if(!string.IsNullOrEmpty(role))
+                sb.Append(root).Append("role:").AppendLine(QuoteIfNeeded(role));
+
+            return ConfigurationFactory.ParseString(sb.ToString());
+        }
+        
+        /// <summary>
+        /// Creates a <see cref="ClusterClient"/> and adds it to the <see cref="ActorRegistry"/> using the given
+        /// <see cref="TKey"/>.
+        /// </summary>
+        /// <param name="builder">The builder instance being configured.</param>
+        /// <param name="initialContacts"> <para>
+        /// List of <see cref="ClusterClientReceptionist"/> <see cref="ActorPath"/> that will be used as a seed
+        /// to discover all of the receptionists in the cluster.
+        /// </para>
+        /// <para>
+        /// This should look something like "akka.tcp://systemName@networkAddress:2552/system/receptionist"
+        /// </para></param>
+        /// <typeparam name="TKey">The key type to use for the <see cref="ActorRegistry"/>.</typeparam>
+        /// <returns>The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.</returns>
+        public static AkkaConfigurationBuilder WithClusterClient<TKey>(
+            this AkkaConfigurationBuilder builder,
+            IList<ActorPath> initialContacts)
```

Comment:
Add an overload that takes `string`  instead of `ActorPath` - just to reduce some upfront friction there.